lib/rules/comma-dangle.js

```diff
@@ -81,49 +81,49 @@ module.exports = {
             category: "Stylistic Issues",
             recommended: false
         },
-
         fixable: "code",
-
-        schema: [
-            {
-                defs: {
-                    value: {
-                        enum: [
-                            "always",
-                            "always-multiline",
-                            "only-multiline",
-                            "never"
-                        ]
-                    },
-                    valueWithIgnore: {
-                        anyOf: [
-                            {
-                                $ref: "#/defs/value"
```

Comment:
Thank you.
Option 2 is a very simple change, I will do it. And I will move fix for comma-dangle in a separate PR.
Option 4 could be a good change for the next major version as users probably expect that all items are validated separately and therefore use $ref's, expecting them to work, rather than ignored. So instead of saying "you can't use $ref if your rule schema is an array" or "you have to understand internal mechanics of eslint to use $ref in such cases" we can just make things work as users expect, either by validating items separately (as users think eslint does) or by transforming schemas correctly taking $ref's into account.